### PR TITLE
refactor: components naming

### LINKED ISSUE #3136: Update component directory and naming convention
- [x] Using our agreed-upon component and directory name convention, go through and improve the current component names
- [x] Document our naming convention

### PR DESCRIPTION
**Pull request by eddybrando**
## Changes

This PR changes the names of our components to align with the conventions that the team agreed upon in a meeting this week.

Closes #3136

### CODE REVIEW

**Review by paaragon — CHANGES_REQUESTED** [review]
I think we should also rename the `public` subfolders. For example `/public/images/landing-page/` -> `/public/images/home/`

**Review by techtolentino — DISMISSED** [review]
@eddybrando this is awesome work

I'm not sure if my feedback should be for _this_ PR, or a separate one.

I'm basically curious if there could be some 'refactoring effort' of simplifying our component names, based on how we organize components. 

In this first review, I use the Advocates card as an example.

File: README.md [hunk 1]
```diff
@@ -101,6 +102,23 @@ GENERATE_CONTENT=1 AIRTABLE_API_KEY=<your airtable api key> npm run dev
 
 <br/>
 
+## 🏷️ Naming Conventions
```
Comment (y4izus) [inline README.md:105]: Really helpful, thank you!

File: components/Advocates/MeetTheAdvocates/AdvocatesMeetTheAdvocatesCard.vue [hunk 2]
```diff
@@ -1,20 +1,23 @@
 <template>
   <UiCard
-    class="advocate-card"
+    class="advocates-meet-the-advocates-card"
```
[the remark this answers is not in the document]
Comment (y4izus) [reply components/Advocates/MeetTheAdvocates/AdvocatesMeetTheAdvocatesCard.vue:3]: I agree, but I also think that refactoring that is out of the scope of this PR. We can create a new issue for talking about this or discussing it offline

File: components/Home/QuickStart/HomeQuickStart.vue [hunk 3]
```diff
@@ -1,16 +1,16 @@
 <template>
-  <article class="cds--grid page-section quick-start">
+  <article class="cds--grid page-section home-quick-start">
```
Comment (y4izus) [inline components/Home/QuickStart/HomeQuickStart.vue:2]: Do we need `QuickStart` here? Maybe `HomeProviders` is enough

File: components/Ui/UiCta.vue [hunk 4]
```diff
@@ -20,7 +20,7 @@
       class="cta__icon"
```
Comment (y4izus) [inline components/Ui/UiCta.vue:20]: Class naming should be change on this file too. Example:

```suggestion
      class="ui-cta__icon"
```

File: pages/ecosystem.vue [hunk 5]
```diff
@@ -1,6 +1,6 @@
 <template>
   <main>
-    <UiPageHeaderFixed class="ecosystem-header__hero">
+    <LayoutLeadSpaceFixed class="ecosystem-header__hero">
```
Comment (y4izus) [inline pages/ecosystem.vue:3]: Should be... ?

```suggestion
    <LayoutLeadSpaceFixed class="ecosystem__header">
```

File: pages/events/summer-school-2022.vue [hunk 6]
```diff
@@ -1,6 +1,6 @@
 <template>
   <main class="summer-school-page">
```
Comment (y4izus) [inline pages/events/summer-school-2022.vue:2]: Should be...? 

```suggestion
  <main class="summer-school-2022-page">
```

File: pages/events/summer-school-2023.vue [hunk 7]
```diff
@@ -1,6 +1,6 @@
 <template>
   <main class="summer-school-page">
```
Comment (y4izus) [inline pages/events/summer-school-2023.vue:2]: Should be...? 

```suggestion
  <main class="summer-school-2023-page">

File: pages/events/index.vue [hunk 8]
```diff
@@ -1,6 +1,6 @@
 <template>
   <div class="event-page">
```
Comment (y4izus) [inline pages/events/index.vue:2]:
```suggestion
  <div class="events-page">
```

**Review by y4izus — CHANGES_REQUESTED** [review]
Awesome refactor!! I've just have some small things :D

**Review by y4izus — APPROVED** [review]

**Review by paaragon — APPROVED** [review]

**Pull request merged**